File: Benchmarks/enterprise/python/fastapi/testcode/benchmark_test_05976.py
# SPDX-License-Identifier: Apache-2.0
from fastapi import Request


def normalise_input(value):
    return value.strip()

async def BenchmarkTest05976(request: Request):
    origin_value = request.headers.get('origin', '')
    data = normalise_input(origin_value)
    globals().setdefault('_secret_cache', {})['current'] = str(data)
    return {"updated": True}

File: Benchmarks/enterprise/java/spring/testcode/BenchmarkTest05976.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import java.io.*;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest05976 {

    private static String normalize(String v) { return v.strip(); }

    @PostMapping(path="/BenchmarkTest05976", consumes="application/xml")
    public void BenchmarkTest05976(@RequestBody String xmlBody, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String xmlValue = xmlBody;
        String data = normalize(xmlValue);
        response.getWriter().print(data + ",data\n");
    }
}
